### PR TITLE
filter habits by user and schedule

### PR DESCRIPTION
**Pull request by MacMalainey**
Habits can now be grabbed by only their respective user and by the day they are to be completed

## Description of Changes

Habits can now be grabbed by only their respective user and by the day they are to be completed
Updated user habit list to only show habits for the given user
Updated daily habit list to only show habits for the given user that are scheduled for that given day

## Screenshots

(Sorry, there should be one here but my screen cap ain't working right)

## Checklist

- [ ] Automated tests
- [ ] Screenshots included (if necessary)
- [ ] Code is well commented

NOTE: habit event tests are broken (will work with owner of those tests to fix)

Closes #ISSUE_ID

### CODE REVIEW

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/models/HabitModel.java [hunk 1]
```diff
@@ -83,9 +95,63 @@ private HabitModel(String id, String title, String reason, Date startDate, Date
...
+                taskManager.setSuccess(models);
+            } else {
+                taskManager.setFailure(task.getException());
+            }
+        });
+        return taskManager.getTask();
+    }
+
+    /**
+     * Get an instance of a habit that exists in the database as specified by the given id
```
Comment (j-vent) [inline src/app/src/main/java/com/cmput301f21t09/budgetprojectname/models/HabitModel.java:133]: nit: make the description different from the function above

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/models/HabitModel.java [hunk 2]
```diff
@@ -83,9 +95,63 @@ private HabitModel(String id, String title, String reason, Date startDate, Date
...
+     *
+     * @return load task for all user's habits
+     */
+    public static ServiceTask<List<HabitModel>> getAllForCurrentUser() {
+        ServiceTaskManager<List<HabitModel>> taskManager = new ServiceTaskManager<>();
+        FirebaseFirestore.getInstance().collection(HABIT_COLLECTION_ID).whereEqualTo("uid", AuthorizationService.getInstance().getCurrentUserId()).get().addOnCompleteListener(task -> {
```
Comment (j-vent) [inline src/app/src/main/java/com/cmput301f21t09/budgetprojectname/models/HabitModel.java:117]: nit: break up long line into multiple lines

**Review by j-vent — APPROVED** [review]
LGTM besides small nits

**Review by Docker-J — APPROVED** [review]

**Review by cheetiantey — APPROVED** [review]
LGTM!

**Review by Docker-J — APPROVED** [review]

**Pull request merged**